modules/apps/currency-converter/currency-converter-web/src/META-INF/resources/view.jsp

```diff
@@ -17,8 +17,17 @@
 <%@ include file="/init.jsp" %>
 
 <%
-String from = ParamUtil.getString(request, "from", CurrencyConverter.DEFAULT_FROM);
-String to = ParamUtil.getString(request, "to", CurrencyConverter.DEFAULT_TO);
+String from = ParamUtil.getString(request, "from");
+
+if (!CurrencyConverterUtil.isCurrency(from)) {
+	from = CurrencyConverter.DEFAULT_FROM;
+}
+
+String to = ParamUtil.getString(request, "to");
+
+if (!CurrencyConverterUtil.isCurrency(to)) {
+	to = CurrencyConverter.DEFAULT_TO;
+}
 
 CurrencyConverter currencyConverter = CurrencyConverterUtil.getCurrencyConverter(from + to);
```

Comment:
If an unsupported currency/string is used, this line returns `null`, which will causes a NPE later in this JSP.